templates/types/streaming/fastapi/app/engine/generate.py

```diff
@@ -0,0 +1,79 @@
+from dotenv import load_dotenv
+
+load_dotenv()
+
+import os
+import logging
+import asyncio
+from llama_index.core.settings import Settings
+from llama_index.core.ingestion import IngestionPipeline
+from llama_index.core.node_parser import SentenceSplitter
+from llama_index.core.storage.docstore import SimpleDocumentStore
+from llama_index.core.storage import StorageContext
+from app.engine.constants import STORAGE_DIR
+from app.settings import init_settings
+from app.engine.loaders import get_documents
+from app.engine.vectordb import get_vector_store
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger()
+
+
+def get_doc_store():
+    if not os.path.exists(STORAGE_DIR):
+        docstore = SimpleDocumentStore()
+        return docstore
+    else:
+        logger.info(f"Loading existing docstore from {STORAGE_DIR}")
+        return SimpleDocumentStore.from_persist_dir(STORAGE_DIR)
```

Comment:
can we just use this line? (as we won't use an in memory docstore)

Reply:
afaik, the ingestion pipeline won't initialize an empty docstore (it will switch to deduplicate strategy if docstore is not provided), and we cannot initialize an empty docstore from an empty storage path.